crates/sui-graphql-rpc/src/types/move_registry/named_type.rs

```diff
@@ -66,8 +67,14 @@ impl NamedType {
         let mut names = vec![];
         let struct_tag = VERSIONED_NAME_UNBOUND_REG.replace_all(name, |m: &regex::Captures| {
             // SAFETY: we know that the regex will always have a match on position 0.
-            names.push(m.get(0).unwrap().as_str().to_string());
-            "0x0".to_string()
+            let name = m.get(0).unwrap().as_str();
+            match VersionedName::from_str(name) {
+                Ok(_) => {
+                    names.push(name.to_string());
+                    "0x0".to_string()
+                }
+                Err(_) => name.to_string(),
+            }
```

Comment:
Interesting -- I'm assuming this was needed before and only noticed now, rather than a new requirement from this change?

```suggestion
            if VersionedName::from_str(name).is_ok() {
                names.push(name.to_string());
                "0x0".to_string()
            } else {
                name.to_string()
            }
```

Reply:
I relaxed the regex a bit, to catch any name-like looking substrings (I want us to be ready for supporting a subname in a SuiNS name too, and it's too complex to regex that), so now I also included the parsing check here. 

Indeed, it might have been better to add this from the beginning.